internal/webhooks/machinedeployment.go

```diff
@@ -268,6 +268,23 @@ func (webhook *MachineDeployment) validate(oldMD, newMD *clusterv1.MachineDeploy
 		}
 	}
 
+	if newMD.Spec.Strategy != nil && newMD.Spec.Strategy.Remediation != nil {
+		total := 1
+		if newMD.Spec.Replicas != nil {
+			total = int(*newMD.Spec.Replicas)
+		}
+
+		if newMD.Spec.Strategy.Remediation.MaxInFlight != nil {
+			if _, err := intstr.GetScaledValueFromIntOrPercent(newMD.Spec.Strategy.Remediation.MaxInFlight, total, true); err != nil {
+				allErrs = append(
+					allErrs,
+					field.Invalid(specPath.Child("strategy", "remediation", "maxInFlight"),
+						newMD.Spec.Strategy.Remediation.MaxInFlight, fmt.Sprintf("must be either an int or a percentage: %v", err.Error())),
```

Comment:
```suggestion
						*newMD.Spec.Strategy.Remediation.MaxInFlight, fmt.Sprintf("must be either an int or a percentage: %v", err.Error())),
```
Maybe? If yes, then same for other cases

Reply:
Not sure if it matters, the pointer has a `String()` method